gcovr/coverage.py

```diff
@@ -213,14 +278,23 @@ def branch_coverage(self):
 
     def decision_coverage(self):
         # type: () -> Tuple[int, int, int, Optional[float]]
-        total = len(self.decisions)
+        total = 0
         cover = 0
-        unchecked = 0
-        for decision in self.decisions.values():
-            if decision.is_covered:
-                cover += 1
-            elif decision.is_uncheckable:
-                unchecked += 1
+        unchecked = False
+        if self.decision is not None:
+            if self.decision.is_uncheckable:
+                unchecked = True
```

Comment:
Not sure how uncheckable decisions should be handled. The comment in the decision analyzer suggests that this decision type is used if there might be *more* than two alternatives. So setting `total = 2` in that case might be reasonable. Must be `total = 1` at least. For comparison, the current state of the HTML writer uses the equivalent of `total = 1`.

Reply:
But it can also be 3, 4 or ... what is the correct value. What about using `number of branches / 2`?
The coverage in % depends on the used value.

Reply:
`2` is a lower bound on the true value. The number of branches is the upper bound on the true value. I think it makes sense (for simplicity) to either offer a most optimistic or most pessimistic estimate of the decision coverage.

With `total = 2` we're choosing the most optimistic estimate. But I think that will be fine, especially since it might still underestimate the true coverage (we count uncheckable decisions as _entirely_ uncovered).

I'm not entirely sure what decision coverage is supposed to represent, since the metric being calculated here is a lot like our branch coverage with all the weird C++-isms stripped out. If decision coverage is supposed to be branch coverage for lines that *look* like they should contain a conditional, then using the branch coverage information for total/covered would allow a more accurate estimation for uncheckable lines.

Not sure what to do here, so I'd rather leave it as it is now (total = 2) and maybe revisit this later.